addons/point_of_sale/static/src/js/barcode_reader.js

```diff
@@ -77,15 +77,15 @@ export const BarcodeReader = core.Class.extend({
      * @param {String} name
      * @param {Function} callback function that takes parsed barcode
      */
-    set_exclusive_callback: function (name, callback) {
+    set_exclusive_callback(name, callback) {
```

Comment:
I wonder if this could just extend EventBus or EventTarget and use standard event mechanics (eg stopImmediatePropagation, maybe combined with capture) instead of having this distinction between exclusive and non-exclusive callbacks and doing all this bookkeeping by hand, especially considering that "exclusive" mode only exists for the login screen.